document-store/src/test/java/org/hypertrace/core/documentstore/query/QueryBuilderTest.java

```diff
@@ -23,6 +45,253 @@ void setUp() {
     queryBuilder = Query.builder();
   }
 
+  @Test
+  public void testQuerySimple() {
+    Query query = Query.builder().build();
+    assertEquals("SELECT * FROM <implicit_collection>", query.toString());
+  }
+
+  @Test
+  public void testQueryWithSelection() {
+    Query query =
+        Query.builder()
+            .addSelection(IdentifierExpression.of("id"))
+            .addSelection(IdentifierExpression.of("fname"), "name")
+            .build();
+    assertEquals("SELECT `id`, `fname` AS name FROM <implicit_collection>", query.toString());
+  }
+
+  @Test
+  public void testQueryWithFilter() {
+    Query query =
+        Query.builder()
+            .setFilter(
+                LogicalExpression.builder()
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("percentage"), GT, ConstantExpression.of(90)))
+                    .operator(AND)
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("class"), EQ, ConstantExpression.of("XII")))
+                    .build())
+            .build();
+
+    assertEquals(
+        "SELECT * FROM <implicit_collection> WHERE (`percentage` > 90) AND (`class` = 'XII')",
```

Comment:
Why are we using this ` one. Can we use a simple quote? If needed can be escaped

Reply:
That is to clearly indicate that it is an Identifier and not a constant.
Constants strings are surrounded with quotes.

Note: Wrapping column names under back ticks (`) is common in some SQL platforms like [MySQL](https://www.tutorialspoint.com/how-to-select-a-column-name-with-spaces-in-mysql#:~:text=To%20select%20a%20column%20name%20with%20spaces%2C%20use%20the%20back,below%20the%20tilde%20operator%20(%20~).)